### PR TITLE
Fixes CASMPET-5032: update image refs for istio charts

### PR DESCRIPTION
**Pull request by bo-quan**
Updated charts to apiVersion v2 and refactored image references to remove dtr.dev.cray.com

### CODE REVIEW

**Review by brantk-hpe — APPROVED** [review]

File: .github/workflows/charts-lint-test-scan.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    uses: Cray-HPE/.github/.github/workflows/charts-lint-test-scan.yml@main
+    with:
+      lint-charts: ${{ github.event_name == 'pull_request' }}
+      ct-config: |
+        chart-dirs:
+          - kubernetes
+        validate-maintainers: false
+      scan-chart-snyk-args: "--severity-threshold=high --policy-path=kubernetes/.snyk"
+      test-charts: false
+      scan-images: false
```
Comment (zcrisler) [inline .github/workflows/charts-lint-test-scan.yml:19]: If we're scanning images nightly (in the charts-lint-test-scan-cron.yml workflow), is there a particular reason why we would not also want to scan them with PRs?

File: kubernetes/cray-istio-deploy/values.yaml [hunk 2]
```diff
@@ -22,7 +22,7 @@ tag: 1.8.6-cray1-distroless
 
 kubectl:
   image:
-    repository: dtr.dev.cray.com/loftsman/loftsman
+    repository: arti.dev.cray.com/csm-docker-stable-local/loftsman
```
Comment (zcrisler) [inline kubernetes/cray-istio-deploy/values.yaml:25]: The appropriate image to use for `kubectl` is `artifactory.algol60.net/csm-docker/stable/docker-kubectl:1.19.15`.

File: kubernetes/cray-istio-operator/values.yaml [hunk 3]
```diff
@@ -4,5 +4,5 @@ istio-operator:
 
 kubectl:
   image:
-    repository: dtr.dev.cray.com/loftsman/loftsman
+    repository: arti.dev.cray.com/csm-docker-stable-local/loftsman
```
Comment (zcrisler) [inline kubernetes/cray-istio-operator/values.yaml:7]: The appropriate image to use for `kubectl` is `artifactory.algol60.net/csm-docker/stable/docker-kubectl:1.19.15`.

File: kubernetes/cray-istio/values.yaml [hunk 4]
```diff
@@ -21,7 +21,7 @@
 # Declare variables to be passed into your templates.
 kubectl:
   image:
-    repository: dtr.dev.cray.com/loftsman/loftsman
+    repository: arti.dev.cray.com/csm-docker-stable-local/loftsman
```
Comment (zcrisler) [inline kubernetes/cray-istio/values.yaml:24]: The appropriate image to use for `kubectl` is `artifactory.algol60.net/csm-docker/stable/docker-kubectl:1.19.15`.

**Review by zcrisler — CHANGES_REQUESTED** [review]

File: .github/workflows/charts-lint-test-scan.yml (hunk 1 above)
Comment (bo-quan) [reply .github/workflows/charts-lint-test-scan.yml:19]: Agree, I'll make it scan with PRs as well in the next patch.

File: kubernetes/cray-istio-deploy/values.yaml (hunk 2 above)
Comment (bo-quan) [reply kubernetes/cray-istio-deploy/values.yaml]: Done

**Comment by zcrisler** [comment]
Looks like `artifactory.algol60.net/csm-docker/stable/istio/proxyv2:1.8.6-cray1-distroless` has some critical and high vulnerabilities. Is there any way we can mitigate them?

File: .github/workflows/charts-lint-test-scan.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+on:
+  pull_request:
+    branches:
+      - master
+      - release/**
+  workflow_dispatch:
+jobs:
+  lint-test-scan:
+    uses: Cray-HPE/.github/.github/workflows/charts-lint-test-scan.yml@main
+    with:
```
Comment (zcrisler) [inline .github/workflows/charts-lint-test-scan.yml:11]: Consider adding `scan-image-snyk-args: "--severity-threshold=high"` here as well.

Comment (bo-quan) [reply .github/workflows/charts-lint-test-scan.yml:11]: Done

**Comment by bo-quan** [comment]
> Looks like `artifactory.algol60.net/csm-docker/stable/istio/proxyv2:1.8.6-cray1-distroless` has some critical and high vulnerabilities. Is there any way we can mitigate them?

@zcrisler As istio upgrade is not trivial, I submitted https://connect.us.cray.com/jira/browse/CASMPET-5166 to track the istio upgrade separately.

**Review by zcrisler — APPROVED** [review]

**Pull request merged**